packages/vhd-lib/src/Vhd/VhdSynthetic.js

```diff
@@ -70,8 +70,8 @@ export class VhdSynthetic extends VhdAbstract {
   async readHeaderAndFooter() {
     await asyncMap(this.#vhds, vhd => vhd.readHeaderAndFooter())
     this.#vhds.forEach((vhd, index) => {
-      if (index < this.#vhds.length) {
```

Comment:
How is this condition required, how can it be not the case?

Reply:
it should have been `length -1` ( which also handles more gracefully the case when there is only one sub vhd). We don't want to force the disktype of the root vhd ( the last one)

Reply:
IMHO, a classic `for` loop would be simpler here :slightly_smiling_face:

Reply:
yes, good idea
Should I also check the chaining as I loop through all of them ?

Reply:
I don't think so because it's difficult to do properly without adding some constraints on the remotes and/or the `VhdAbstract` instances that can be passed.

Also, maybe we could do this using only the `parentUuid`?